### ISSUE TITLE
Better dataset path management

### ISSUE THREAD

**Issue by Kirill888** [opendatacube/datacube-core#586]
### Expected behaviour

```bash
cd /g/data/some/path/
datacube dataset add some.yaml
```
Expect new dataset to be added with location pointing to `/g/data/some/path/some.yaml`.

### Actual behaviour

Location is set to "actual" path `/g/data2/some/path/some.yaml`. In this case `/g/data/some` is a symlink to `/g/data2/some`.

raised by @jeremyh in https://github.com/GeoscienceAustralia/digitalearthau/issues/179

### How to fix

We currently use Python's `pathlib.Path(some_path).absolute()` to get path to be indexed. This is however problematic fo two reasons

1. The symlink info is lost
2. If path is like this `../some/path.txt` it will not be normalised (will still have `..` in it)

The only way to get directory name as user sees it is via environment variable "PWD" (if it is set, if not use current approach). There is however issue of process possibly calling `os.chdir(..)` at some point in which case `PWD` and actual woking directory will not be the same thing, so we should probably test for that.

We also need to call `os.path.normpath` to remove any `../` and `./` is present in the path.

**Comment by omad** [comment]
I can't find a `pathlib.Path(path).absolute()` method, but think we should be using `pathlib.Path().resolve()` which returns an absolute path and strips out `..` and `.` elements.

I think trying to maintain symlinks is just going to be error prone and problematic.

**Comment by Kirill888** [comment]
@omad happens in several places
```
datacube_apps/stacker/fixer.py:172:    output_uri = output_filename.absolute().as_uri()
datacube_apps/stacker/stacker.py:149:    output_uri = output_filename.absolute().as_uri()
datacube/utils/__init__.py:391:                return path.absolute().as_uri(), doc
datacube/utils/__init__.py:395:                return mk_part_uri(path.absolute().as_uri(), idx), doc
datacube/scripts/ingest.py:238:            return file_path.absolute().as_uri()
```

**Comment by Kirill888** [comment]
EDIT:

we can use `os.path.normpath` to get rid of `..` without touching symlink info.

OLD:
>by the way `resolve` doesn't undo symlinks, neither does `absolute` when operating on already >absolute path, only `absolute()` when mapping from relative to absolute.

**Comment by Kirill888** [comment]
meant to say `os.path.resolve` doesn't, not sure about `Path().resolve`

**Comment by Kirill888** [comment]
@omad how's this

https://github.com/opendatacube/dea-proto/blob/305dae74ba4b5454362ee80da968ed34f473002f/dea/io/path.py

seems to work on my mac

**Comment by omad** [comment]
Resolved in #587
